### ISSUE TITLE
Broken source code link from `cljdoc.org` to `github.com`

### ISSUE THREAD

**Issue by cloojure** [cljdoc/cljdoc#753]
Suppose I navigate to documentation for `tupelo.core/truthy?` here:

https://cljdoc.org/d/tupelo/tupelo/23.03.03c-SHAPSHOT/api/tupelo.core#truthy?

When I click on the link for `source`, I end up at the GitHub page here:

https://github.com/cloojure/tupelo/blob/v23.03.03c-SHAPSHOT/#L293

which returns:   404 "This is not the webpage you are looking for".  

It appears that GitHub needs a URL formatted like:

https://github.com/cloojure/tupelo/blob/v23.03.03c-SHAPSHOT/src/cljc/tupelo/core.cljc#L2873

or similar.

**Comment by lread** [comment]
Hmmm... interesting, thanks for reporting @cloojure.
I'll take a peek sometime soon-ish.

**Comment by lread** [comment]
Does not seem specific to SNAPSHOT builds.
I see the same issue here: https://cljdoc.org/d/tupelo/tupelo/22.08.03/api/tupelo.core

**Comment by cloojure** [comment]
Yes, I first noticed it 6+ months ago, but didn't follow up at the time.  I
just switched from lein to deps/cli and thought I should figure out what
was going wrong.

Possibly GitHub has changed their API.

Alan

**Comment by lread** [comment]
Turns out that it is a cljdoc problem, not a GitHub one.

I'm looking at tupelo/tupelo 22.08.03.

At doc build time cljdoc analyzes the jar sources.

At render time, for the "source" link, it matches the jar source files back to the GitHub source files.
If it only sees one match, no problemo, it links to that.
For `tupelo/core.cljc` it is seeing two candidates on GitHub:
- `test/cljc/tst/tupelo/core.cljc`
- `src/cljc/tupelo/core.cljc`
And is currently seeing neither as a good choice, hence the broken link.

The cljdoc algorithm currently favours the file with the shortest source path ending in `/src/` or `/src/main/`, but does not consider `/src/cljc/`.

I'll tweak the algorithm. I think favouring the file with the shortest source path that has a `src` dir would be an improvement.

**Comment by cloojure** [comment]
Cool. Seems to be working again.

**Comment by lread** [comment]
Awesome! Thanks for raising the issue, by doing so you helped to improve cljdoc!
